components/layout_2020/text.rs

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use gfx::text::glyph::GlyphStore;
+use std::sync::Arc;
+use style::values::computed::Length;
+use style::Zero;
+use webrender_api::FontInstanceKey;
+
+#[derive(Clone)]
+pub struct ShapedSegment {
+    pub(crate) font_key: FontInstanceKey,
+    pub(crate) glyph_runs: Vec<Arc<GlyphStore>>,
```

Comment:
"Segment" and "run" are somewhat interchangeable in my mind. What’s the terminology here? What does it mean to have multiple runs per segment? Since there is only one font key per segment, I assume we also have multiple segments per DOM text node.

Reply:
> multiple segments per DOM text node

Though maybe not yet if we don’t do font fallback? Either way, could we unify the segment and (glyph) run concepts with each other? (And maybe with `flow::inline::TextRun`?)

Reply:
Removing an unnecessary abstraction sounds great!